### PR TITLE
Add itemtype and item to sockets table

### PR DESCRIPTION
**Pull request by rbrannath**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #13214

Adds the item type and item of a socket to the socket table of an item, which is defined via the cable.

### CODE REVIEW

File: src/Socket.php [hunk 1]
```diff
@@ -798,6 +800,28 @@ public static function showListForItem($item)
                 echo "<td></td>";
             }
 
+            if (
+                $cable->fields['itemtype_endpoint_a'] === $item->getType() &&
+                    $cable->fields['items_id_endpoint_a'] === $item->getID()
+            ) {
+                    $itemtype = $cable->fields['itemtype_endpoint_b'];
+                    $itemId = $cable->fields['items_id_endpoint_b'];
+            } else {
```
Comment (cedric-anne) [inline src/Socket.php:809]:
```suggestion
            if (
                $cable->fields['itemtype_endpoint_a'] === $item->getType()
                && $cable->fields['items_id_endpoint_a'] === $item->getID()
            ) {
                $itemtype = $cable->fields['itemtype_endpoint_b'];
                $itemId = $cable->fields['items_id_endpoint_b'];
            } else {
```

File: src/Socket.php [hunk 2]
```diff
@@ -798,6 +800,28 @@ public static function showListForItem($item)
                 echo "<td></td>";
             }
 
+            if (
+                $cable->fields['itemtype_endpoint_a'] === $item->getType() &&
+                    $cable->fields['items_id_endpoint_a'] === $item->getID()
+            ) {
+                    $itemtype = $cable->fields['itemtype_endpoint_b'];
+                    $itemId = $cable->fields['items_id_endpoint_b'];
```
Comment (cedric-anne) [inline src/Socket.php:808]:
```suggestion
                    $item_id = $cable->fields['items_id_endpoint_b'];
```

File: src/Socket.php [hunk 3]
```diff
@@ -798,6 +800,28 @@ public static function showListForItem($item)
...
+            } else {
+                $itemtype = $cable->fields['itemtype_endpoint_a'];
+                $itemId = $cable->fields['items_id_endpoint_a'];
+            }
+
+
+            if ($itemId !== 0) {
+                $endpoint = new $itemtype();
+                $endpoint->getFromDB($itemId);
```
Comment (cedric-anne) [inline src/Socket.php:817]:
```suggestion
            $endpoint = getItemForItemtype($itemtype);
            if ($endpoint !== false && $item_id !== 0 && $endpoint->getFromDB($item_id)) {
```

File: src/Socket.php [hunk 4]
```diff
@@ -798,6 +800,26 @@ public static function showListForItem($item)
...
+            if (
+                $cable->fields['itemtype_endpoint_a'] === $item->getType() &&
+                    $cable->fields['items_id_endpoint_a'] === $item->getID()
+            ) {
+                    $itemtype = $cable->fields['itemtype_endpoint_b'];
+                    $item_id = $cable->fields['items_id_endpoint_b'];
+            } else {
+                $itemtype = $cable->fields['itemtype_endpoint_a'];
+                $itemId = $cable->fields['items_id_endpoint_a'];
+            }
```
Comment (cedric-anne) [inline src/Socket.php:812]:
```suggestion
            if (
                $cable->fields['itemtype_endpoint_a'] === $item->getType()
                && $cable->fields['items_id_endpoint_a'] === $item->getID()
            ) {
                $itemtype = $cable->fields['itemtype_endpoint_b'];
                $item_id = $cable->fields['items_id_endpoint_b'];
            } else {
                $itemtype = $cable->fields['itemtype_endpoint_a'];
                $item_id = $cable->fields['items_id_endpoint_a'];
            }
```

**Review by cedric-anne — APPROVED** [review]

**Comment by trasher** [comment]
Should not this one target 10.0/bf?

**Comment by cedric-anne** [comment]
Picked in 10.0/bugfixes branch (a930ef2f09d444a4d20b2b1c04a5a94cc275ec71).

Thanks for the contribution.

**Pull request closed without merging**